shell/common/pointer_data_dispatcher.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/common/pointer_data_dispatcher.h"
+
+namespace flutter {
+
+PointerDataDispatcher::~PointerDataDispatcher() = default;
+DefaultPointerDataDispatcher::~DefaultPointerDataDispatcher() = default;
+SmoothPointerDataDispatcher::~SmoothPointerDataDispatcher() = default;
+
+void DefaultPointerDataDispatcher::DispatchPacket(
+    std::unique_ptr<PointerDataPacket> packet,
+    uint64_t trace_flow_id) {
+  delegate_.DoDispatchPacket(std::move(packet), trace_flow_id);
+}
+
+void SmoothPointerDataDispatcher::DispatchPacket(
+    std::unique_ptr<PointerDataPacket> packet,
+    uint64_t trace_flow_id) {
+  if (is_pointer_data_in_progress_) {
+    if (pending_packet_ != nullptr) {
+      DispatchPendingPacket();
```

Comment:
Won't this possibly make SmoothPointerDataDispatcher not work?  If we call this at any frequency greater than 30hz (nyquist of framerate) we will be sending packets outside of the vsync time.

Reply:
I'm not sure why you think it won't work. We can have a discussion offline. Anyway, this is unchanged in this revert/reland.